### PR TITLE
Fix #10220 - LifeCycleCost objects are using "Steam" not "DistrictHeatingSteam"

### LINKED ISSUE #10220: LifeCycleCost objects are using "Steam" not "DistrictHeatingSteam"
Issue overview
--------------

Seems that generally the fuel type `Steam` was replaced with `DistrictHeatingSteam`.

`LifeCycleCost:UsePriceEscalation` and `LifeCycleCost:UseAdjustment `use `Steam`. 

Also, they, and `FuelFactors` as well, don't include `DistrictHeatingWater` nor `DistrictCooling`, is that normal?

`Exterior:FuelEquipment` has "Water" in the list of fuel types, it shouldn't. There is `Exterior:WaterEquipment` for that
   *   It throws if you use it actually: https://github.com/NREL/EnergyPlus/blob/281383d95af45331e9c1fba41fa1dba5d66a59c5/src/EnergyPlus/InternalHeatGains.cc#L2674-L2677

Here is a spreadsheet showing the current status in the entire IDD (assuming my parsing is correct), with comments

https://docs.google.com/spreadsheets/d/1ZjLABsEOAYmAyrrI6iM1JMJi6uRYBMWHlu3AUcx40iE/edit#gid=518449229

Pasting an image for convenience, note that this may get outdated:

image

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version): all
 - Version of EnergyPlus (if using an intermediate build, include SHA): V23.2.0-IOFreeze

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Defect file added (list location of defect file here)
 - [x] Ticket added to Pivotal for defect (development team task)
 - [ ] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by jmarrec**
Pull request overview
---------------------
 - Fixes #10220
 - Transition for Steam => DistrictHeatingSteam for LCC objects
 - Add DistrictHeating and DistrictCooling as well

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [x] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [x] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [x] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [x] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [x] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: idd/Energy+.idd.in [hunk 1]
```diff
@@ -100940,7 +100940,9 @@ LifeCycleCost:UsePriceEscalation,
         \key ElectricitySurplusSold
         \key ElectricityNet
         \key NaturalGas
-        \key Steam
+        \key DistrictCooling
+        \key DistrictHeatingWater
+        \key DistrictHeatingSteam
```
Comment (jmarrec) [inline idd/Energy+.idd.in:100945]: For `LifeCycleCost:UsePriceEscalation` and `LifeCycleCost:UseAdjustment`

Replace `Steam` with `DistrictHeatingSteam` like the rest of the objects

File: src/Transition/CreateNewIDFUsingRulesV23_2_0.f90 [hunk 2]
```diff
@@ -916,6 +916,22 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
...
+                     OutArgs(2) = "DistrictHeatingSteam"
+                 END IF
+
+              CASE('LIFECYCLECOST:USEADJUSTMENT')
+                 CALL GetNewObjectDefInIDD(ObjectName,NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                 nodiff=.false.
+                 OutArgs(1:CurArgs)=InArgs(1:CurArgs)
+                 IF (SameString(OutArgs(2), "STEAM")) THEN
+                     OutArgs(2) = "DistrictHeatingSteam"
+                 END IF
```
Comment (jmarrec) [inline src/Transition/CreateNewIDFUsingRulesV23_2_0.f90:933]: Transition. See test on DevSupport: https://github.com/NREL/EnergyPlusDevSupport/blob/master/DefectFiles/10000s/10220/transition.idf

```
./Transition-V23-1-0-to-V23-2-0 /path/to/EnergyPlusDevSupport/DefectFiles/10000s/10220/transition.idf
gvim -d -f transition.idfold transition.idfnew
```

image

File: tst/EnergyPlus/unit/EconomicLifeCycleCost.unit.cc [hunk 3]
```diff
@@ -585,3 +588,69 @@ TEST_F(EnergyPlusFixture, EconomicLifeCycleCost_ExpressAsCashFlows)
...
+
+    EXPECT_EQ(numResources, state->dataEconLifeCycleCost->numUsePriceEscalation);
+    for (const auto &lcc : state->dataEconLifeCycleCost->UsePriceEscalation) {
+        EXPECT_FALSE(compare_enums(lcc.resource, Constant::eResource::Invalid, false)) << "Failed for " << lcc.name;
+    }
+    EXPECT_EQ(numResources, state->dataEconLifeCycleCost->numUseAdjustment);
+    for (const auto &lcc : state->dataEconLifeCycleCost->UseAdjustment) {
+        EXPECT_FALSE(compare_enums(lcc.resource, Constant::eResource::Invalid, false)) << "Failed for " << lcc.name;
+    }
+}
```
Comment (jmarrec) [inline tst/EnergyPlus/unit/EconomicLifeCycleCost.unit.cc:656]: (meta) testing to ensure this doesn't creep back in.

**Comment by jmarrec** [comment]
Using the MCVE files on DevSupport: 

* in_ori.idf: uses `Steam` LCC objects, and does not include `DistrictCooling`, `DistrictHeatingWater` or it throws on input parsing
* in.idf uses `DistrictHeatingSteam`, `DistrictCooling`, `DistrictHeatingWater` objects
* Both files include LCC objects for all other fuels

## Ori: 

eplusout.err has a warning

```
   ** Warning ** The resource referenced by LifeCycleCost:UsePriceEscalation= "LCCUSEPRICEESCALATION DISTRICTHEATINGSTEAM" has no energy cost. 
   **   ~~~   ** ... It is likely that the wrong resource is used. The resource should match the meter used in Utility:Tariff.
```
as you can see, DistrictHeatingSteam is not changed whatsoever.

image

## New:


All fuels are properly affected by the LCC objects:

image

File: src/Transition/CreateNewIDFUsingRulesV23_2_0.f90 (hunk 2 above)
Comment (Myoldmopar) [inline src/Transition/CreateNewIDFUsingRulesV23_2_0.f90:933]: Great catch, @jmarrec, thanks.

File: idd/Energy+.idd.in (hunk 1 above)
Comment (Myoldmopar) [reply idd/Energy+.idd.in:100945]: OK, good catch.

**Review by Myoldmopar — APPROVED** [review]

**Comment by Myoldmopar** [comment]
This also tests happily, and the only diff is the spurious table diff.  I am going to go ahead and merge this in, thanks @jmarrec

**Pull request merged**